### PR TITLE
Add Math translators for Cosmos DB provider

### PR DESCRIPTION
**Pull request by Marusyk**
Added Math for Cosmos:
ABS, CEILING, FLOOR, POWER, EXP, LOG10, LOG, SQRT, ACOS, ASIN, ATAN, ATN2, COS, SIN, TAN, SIGN, TRUNC, ROUND

Issue: #16143

Need your help with translation methods with converting.
Example:

```
.Where(od => Math.Tan(od.Discount) > 0)
```
here `od.Discount` is `float`. `Math.Tan` requires `double`. So, the following expression will be generated:

```
Tan(Convert([Microsoft.EntityFrameworkCore.Query.EntityShaperExpression].Discount, Double))
```
In this case the `TryRemoveImplicitConvert` will be called and return `null`

https://github.com/Marusyk/efcore/blob/e23cdc0128851a8760bbcd57696f4001ae3b3a60/src/EFCore.Cosmos/Query/Internal/CosmosSqlTranslatingExpressionVisitor.cs#L685

Could you please suggest how to improve that?

### CODE REVIEW

**Comment by Marusyk** [comment]
any updates about this PR?

**Comment by bricelam** [comment]
@maumar I'm poaching this.

**Comment by bricelam** [comment]
@Marusyk Interested in adding the `MathF` versions too? (like PR #24829) If not, I'll add them while merging...

**Comment by Marusyk** [comment]
@bricelam sure, I will add

**Comment by bricelam** [comment]
@maumar Looks like this still needs your expertise to answer some design questions.

**Comment by Marusyk** [comment]
I will add tests after merge of #18842

**Comment by Marusyk** [comment]
any updates?

File: src/EFCore.Cosmos/Query/Internal/MathTranslator.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            Check.NotNull(arguments, nameof(arguments));
+            Check.NotNull(logger, nameof(logger));
+
+            if (_supportedMethodTranslations.TryGetValue(method, out var sqlFunctionName))
+            {
+                var typeMapping = arguments.Count == 1
+                    ? ExpressionExtensions.InferTypeMapping(arguments[0])
+                    : ExpressionExtensions.InferTypeMapping(arguments[0], arguments[1]);
+
+                var newArguments = new SqlExpression[arguments.Count];
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Query/Internal/MathTranslator.cs:114]: You can just use
`newArguments = arguments.Select(e => _sqlExpressionFactory.ApplyTypeMapping(e, typeMapping!)` 
and inline newArguments variable.

**Comment by smitpatel** [comment]
@Marusyk - For now disable tests which are running into issue of convert node. Looking at the CosmosDb docs, seems like they only have double kind of numeric values. This would make any convert to double node in LINQ for numeric type no-op and translatable but I need to research more on this in order to update conversions.

File: src/EFCore.Cosmos/Query/Internal/CosmosSqlTranslatingExpressionVisitor.cs [hunk 2]
```diff
@@ -710,6 +710,30 @@ private static Expression TryRemoveImplicitConvert(Expression expression)
                 }
             }
 
+            if (expression is MethodCallExpression methodCallExpression)
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Query/Internal/CosmosSqlTranslatingExpressionVisitor.cs:713]: Remove this code block.

**Review by smitpatel — APPROVED** [review]

**Comment by smitpatel** [comment]
@Marusyk - 1 test is failing in SQL assertion (perhaps not getting translated now), can you also disable it and update the PR?

**Comment by Marusyk** [comment]
@smitpatel I've fixed it. Thanks

**Comment by smitpatel** [comment]
@Marusyk - Thank you for contribution.

**Pull request merged**